### PR TITLE
fix: add batch_no to quotation item, add batch_no to update_cart function, pass batch_no from shopping cart

### PR DESCRIPTION
**Pull request by rvpasari**
https://app.asana.com/0/1192407975826276/1199647953467852

### CODE REVIEW

**Comment by rvpasari** [comment]
connected to: 

https://github.com/DigiThinkIT/humboldt/pull/20

File: erpnext/stock/doctype/batch/batch.py [hunk 1]
```diff
@@ -414,8 +470,13 @@ def get_active_batch(item_code):
 				"certificate_of_analysis": "/files/coa.pdf"
 			}
 	"""
-
 	fields = ["name", "item", "item_name", "stock_uom", "thc", "cbd", "certificate_of_analysis", "batch_qty"]
 	active_batch = frappe.get_all("Batch", filters={"item": item_code, "display_on_website": 1}, fields=fields)
```
Comment (hrwX) [inline erpnext/stock/doctype/batch/batch.py]:
```suggestion
	active_batch = frappe.get_all("Batch", filters={"item": item_code, "display_on_website": 1}, fields=fields, limit=1)
```

File: erpnext/stock/doctype/batch/batch.py [hunk 2]
```diff
@@ -393,6 +393,62 @@ def get_readings_for_qi(qi_name):
...
+						and delivered_by_supplier = 0
+					) as so_item_delivered_qty,
+					parent, name
+				from
+				(
+					select qty, parent_detail_docname, parent, name
+					from `tabPacked Item` dnpi_in
+					where item_code = %s and warehouse = %s
+					and parenttype="Sales Order"
+					and item_code != parent_item
```
Comment (hrwX) [inline erpnext/stock/doctype/batch/batch.py]: - Rename to `dnpi` to something better
- Prepend table name to fields, there are chances that mysql might throw ambiguous error .

File: erpnext/shopping_cart/cart.py [hunk 3]
```diff
@@ -145,10 +145,11 @@ def update_cart(item_code, qty, row=None, additional_notes=None, with_items=Fals
 	else:
 		quotation_items = quotation.get("items", {"item_code": item_code})
 		if not quotation_items:
-			quotation.append("items", {
+			quotation.append("items", {
```
Comment (hrwX) [inline erpnext/shopping_cart/cart.py:148]:
```suggestion
			quotation.append("items", {
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/batch/batch.py (hunk 1 above)
Comment (sahil28297) [reply erpnext/stock/doctype/batch/batch.py]: If you want to fetch only the first value, use `get_value` instead of `get_all`

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/batch/batch.py (hunk 2 above)
Comment (rvpasari) [reply erpnext/stock/doctype/batch/batch.py]: I've just copied this from the reserve quantity function for stock without batches.

Comment (sahil28297) [reply erpnext/stock/doctype/batch/batch.py]: Then you'll need to fix it over there as well. Just add what is `dnpi_qty` to the docstring above, its okay to use acronyms in large statements but it should be defined somewhere.

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**